runner/src/main/java/org/ananas/runner/model/steps/commons/execution/PipelineOptionsFactory.java

```diff
@@ -9,40 +10,54 @@
 
 public class PipelineOptionsFactory {
 
-	public static PipelineOptions create(boolean isTest) {
-		return isTest ? createFlinkOptions() : createSparkOptions();
+	public static PipelineOptions create(boolean isTest, DagRequest.Engine engine) {
+		if (isTest) {
+			return createFlinkOptions(null);
+		}
+
+		switch(engine.type) {
+			case "Flink":
+				return createFlinkOptions(engine);
+			case "Spark":
+				return createSparkOptions(engine);
+			default:
+				return createFlinkOptions(null);
+		}
 	}
 
 
-	public static PipelineOptions createFlinkOptions() {
+	public static PipelineOptions createFlinkOptions(DagRequest.Engine engine) {
 		FlinkPipelineOptions options =
 				org.apache.beam.sdk.options.PipelineOptionsFactory.create().as(FlinkPipelineOptions.class);
-		options.setParallelism(10);
-		options.setMaxBundleSize(1000 * 1000L);
-		options.setObjectReuse(true);
-
+		if (engine == null) {
+			options.setParallelism(10);
+			options.setMaxBundleSize(1000 * 1000L);
+			options.setObjectReuse(true);
+		} else {
+			options.setParallelism(engine.getProperty("parallelism", Integer.valueOf(10)));
```

Comment:
I believe we need to set engine specific parameters here

Reply:
This one is the engine specific parameters for Flink.

Reply:
This part is for Flink specific parameters. As currently Flink is only used for the local job execution,  I didn't put the parameters for clusters. We can add these cluster related parameters in a separate ticket for Flink cluster support.